slick/src/main/scala/slick/jdbc/StatementInvoker.scala

```diff
@@ -30,7 +30,8 @@ abstract class StatementInvoker[+R] extends Invoker[R] { self =>
              (implicit session: JdbcBackend#Session): Either[Int, PositionedResultIterator[R]] = {
     //TODO Support multiple results
     val statement = getStatement
-    val st = session.prepareStatement(statement, defaultType, defaultConcurrency, defaultHoldability)
+    val fetchSizeOverride = if (maxRows == 1) Some(1) else None
```

Comment:
What would be the right `fetchSizeOverride` if `maxRows > 1`?

Reply:
To be perfectly honest fetchSizeOverride is used just once in whole Slick codebase: StatementInvoker.scala#34: 
```
val fetchSizeOverride = if (maxRows == 1) Some(1) else None
val st = session.prepareStatement(statement, defaultType, defaultConcurrency, defaultHoldability, fetchSizeOverride)
```
This also reveals the intent of this change - I only override the fetch size when maxRows is equal one, so for single row fetches. I recall that for those cases the `fetchSize` param was set to 0 in JdbcBackend impls and that caused the issue of leaving portals unclosed. Setting fetchSize to actual amount of rows that you want to get allows for paging of ResultSet I think and if fetchSize == maxRows it just fetches that amount of rows and closes the portal immediately afterwards. That's the behaviour that we actually want, because it's a sane resource handling practice that was left ignored because many databases close portals implicitly on transaction commit/rollback or connection close. Cockroach doesn't as it would incur consensus / network roundtrip AFAIU, so we have to issue the portal close command manually. 

I've made `fetchSizeOverride` an `Option[Int]` because maybe somebody else will find a use case where they want to fine-tune it on `prepareStatement` step and they don't want it to be just set to one. For my use case it could be just a boolean flag that, if set to true, sets fetchSize to 1 on the statement. 

I'm also making the default arg for `fetchSizeOverride` `None`.

Reply:
I just realized that I provided context but didn't answer your question straightforward: there's no right value for `fetchSizeOverride` for `maxRows > 1` as this override is used right now only to close portals in single-row fetch queries. `fetchSize` itself is set in `JdbcBackend` impls and it seems that it gets set correctly. It just doesn't for `maxRows == 1`.